#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: bash-4.0 upstream fix 032

			     BASH PATCH REPORT
			     =================

Bash-Release: 4.0
Patch-ID: bash40-032

Bug-Reported-by:	muszi@muszi.kite.hu
Bug-Reference-ID:	<20090826113159.18815.qmail@muszi.kite.hu>
Bug-Reference-URL:	http://lists.gnu.org/archive/html/bug-bash/2009-08/msg00090.html

Bug-Description:

Bash-4.0 has a memory leak when processing ${!prefix@}.

Patch:

*** ../bash-4.0-patched/subst.c	2009-07-22 23:18:55.000000000 -0400
--- subst.c	2009-08-26 23:08:51.000000000 -0400
***************
*** 6607,6611 ****
  	}
        free (x);
!       free (xlist);
        free (temp1);
        *indexp = sindex;
--- 6769,6773 ----
  	}
        free (x);
!       dispose_words (xlist);
        free (temp1);
        *indexp = sindex;
*** ../bash-4.0/patchlevel.h	2009-01-04 14:32:40.000000000 -0500
--- patchlevel.h	2009-02-22 16:11:31.000000000 -0500
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 31
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 32
  
  #endif /* _PATCHLEVEL_H_ */
